src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -37,8 +37,10 @@ internal interface IWorkspaceProjectContext : IDisposable
         // Files.
         void AddSourceFile(string filePath, bool isInCurrentContext = true, IEnumerable<string> folderNames = null, SourceCodeKind sourceCodeKind = SourceCodeKind.Regular);
         void RemoveSourceFile(string filePath);
+        void RenameSourceFile(string originalFilePath, string newFilePath);
         void AddAdditionalFile(string filePath, bool isInCurrentContext = true);
         void RemoveAdditionalFile(string filePath);
+        void RenameAdditionalFile(string originalFilePath, string newFilePath);
         void AddDynamicFile(string filePath, IEnumerable<string> folderNames = null);
         void RemoveDynamicFile(string filePath);
```

Comment:
What about rename of DynamicFile?

Reply:
Currently dynamic files don't support this from a deeper infrastructure level. I'm also not 100% sure what dynamic files are. What's the use case for them? If needed, I can wire them up too

Reply:
Razor files. It would be preferable that these have the same shape from our perspective and if you want to model them differently (say via a Remove then Add) under the covers we don't need to know about it. It just lets us propagate renames for all files through to you and treat them the same.

Reply:
Got it, I'll add dynamic file support in the API then